src/scrabble.js

```diff
@@ -1,6 +1,98 @@
 class Scrabble {
-  // Write your implementation here
+  constructor (word) {
+    this.word = word
+  }
 
+  scoreLoop (string) {
+    let num = 0
+    for (let i = 0; i < string.length; i++) {
+      if (string[i].match(/[^A-Z{}[]]/i)) {
+        return 0
+      }
+      if (string[i].match(/[AEIOULNRST]/i)) {
+        num++
+      }
+      if (string[i].match(/[DG]/i)) {
+        num += 2
+      }
+      if (string[i].match(/[BCMP]/i)) {
+        num += 3
+      }
+      if (string[i].match(/[FHVWY]/i)) {
+        num += 4
+      }
+      if (string[i].match(/[K]/i)) {
+        num += 5
+      }
+      if (string[i].match(/[JX]/i)) {
+        num += 8
+      }
+      if (string[i].match(/[QZ]/i)) {
+        num += 10
+      }
+    }
+    return num
+  }
+
+  scoreMain () {
+    if (this.word == null) {
+      return 0
+    }
+
+    const wordTrimmed = this.word.trim()
+
+    return this.scoreLoop(wordTrimmed)
+  }
+
+  scoreDoubleTripleLetter (open, close) {
+    if (this.word == null) {
+      return 0
+    }
+
+    if (this.word.indexOf(open) === -1 && this.word.indexOf(close) === -1) {
+      return 0
+    }
+
+    const letterArray = []
+
+    for (let i = 0; i < this.word.length; i++) {
+      if (this.word[i + 2] !== undefined) {
+        if (this.word[i] === open && this.word[i + 2] === close) {
+          letterArray.push(this.word[i + 1])
+        }
+      }
+    }
+
+    if (letterArray.length === 0) {
```

Comment:
Looking at the code below this line, I don't think this is required. If the array is empty, your for-loop wont run.  so the `score` variable will be set to 0 when it is returned.

Reply:
My thinking was that I wanted to stop the for loop running as it would not be efficient. But I didn't think about it not running when the length was 0 anyway! Good to know!